### PR TITLE
Python 3 + New display options

### PR DESCRIPTION
**Pull request by rgerkin**

### CODE REVIEW

**Comment by niklasf** [comment]
Hi @rgerkin, thanks for your pull request!

* I think it would be best to keep Python 3 compability fixes and new features seperated.
* Python 3 is generally already supported.

I'll provide some more specific feedback as comments to the commits.

Niklas

File: chess/__init__.py [hunk 1]
```diff
@@ -26,12 +26,17 @@
 import re
 
 
-COLORS = [ WHITE, BLACK ] = range(2)
+COLORS = [ WHITE, BLACK ] = list(range(2))
```
Comment (niklasf) [inline chess/__init__.py]: In general adding list(...) around ranges or views is not required. We're just iterating over them anyway. This should also apply to all other cases.

File: chess/uci.py [hunk 2]
```diff
@@ -16,8 +16,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (niklasf) [inline chess/uci.py]: This is a valid cleanup.

File: chess/uci.py [hunk 3]
```diff
@@ -29,7 +29,7 @@
 try:
     import queue
 except ImportError:
-    import Queue as queue
+    import queue as queue
```
Comment (niklasf) [inline chess/uci.py]: The uppercase `Queue` is intended as a fallback for Python 2.

File: chess/polyglot.py [hunk 4]
```diff
@@ -158,7 +158,7 @@ def next_raw(self):
         except struct.error:
             raise StopIteration()
 
-    def next(self):
```
Comment (niklasf) [inline chess/polyglot.py]: I think this rename is correct. We should keep `next` as an alias for Python 2.

File: chess/__init__.py [hunk 5]
```diff
@@ -2418,6 +2426,60 @@ def __str__(self):
 
         return "".join(builder)
 
+    def __unicode__(self):
+        string = '   a  b  c  d e  f  g  h  \r\n'
+        for rank in range(8,0,-1):
+            string += '  ' + '-'*22 + '\r\n'
+            string += '%d ' % rank
+            for file_ in 'A B C D E F G H'.split():
+                square = eval('%s%d' % (file_,rank))
```
Comment (niklasf) [inline chess/__init__.py]: Let's not use `eval`.

**Comment by niklasf** [comment]
Maybe a good next stop would be getting in the new `__unicode__()`.

**Comment by rgerkin** [comment]
This pull request is now obsolete.

**Pull request closed without merging**